### PR TITLE
Capture dependencies groups with inline comments in pyproject files

### PR DESCRIPTION
**Pull request by dsuleimenov**
Fix for the following issue:
https://github.com/dependabot/dependabot-core/issues/8366

### CODE REVIEW

File: python/spec/dependabot/python/file_updater/poetry_file_updater_spec.rb [hunk 1]
```diff
@@ -501,6 +501,137 @@
...
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["dev"]
+            }, {
+              requirement: ">=3.8.2",
+              file: "pyproject.toml",
+              source: nil,
+              groups: ["test"]
+            }]
+          )
```
Comment (deivid-rodriguez) [inline python/spec/dependabot/python/file_updater/poetry_file_updater_spec.rb:544]: In real life, all `requirements` here will be for the same dependency, so to better represent real behavior, each dependency should configure the requirements array for the dependency it's testing (similarly to how you customized `dependency_name`. Does that make sense? Basically, I think you'd need to add a `let(:requirements)` to each spec, and use that here.  Does that make sense?

**Review by deivid-rodriguez — COMMENTED** [review]
This is great, just left a minor comment on the specs!

File: python/spec/dependabot/python/file_updater/poetry_file_updater_spec.rb (hunk 1 above)
Comment (dsuleimenov) [reply python/spec/dependabot/python/file_updater/poetry_file_updater_spec.rb]: Certainly, I agree with your suggestion.
Thanks for the review, done 👍

**Comment by dsuleimenov** [comment]
Hi @deivid-rodriguez
Could you please review my PR again when you have time?

**Comment by deivid-rodriguez** [comment]
Thanks so much for the fix! This is now deployed to the Github service, let us know if you run into any issues!

**Pull request merged**